### ISSUE TITLE
Discovery spike: Adding "verify" alerts to sign-in widget

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-team#88547]
### User story

As a Cartography team member, I want to document our approach for adding a sign-in widget to the supply reordering page that includes verification alerts, so that we can be sure of how to proceed in a way that supports similar widgets going forward.

### Notes
- Sign-in widget work was started in #79823
- Consult with CAIA/Danielle Thierry, there are likely sign-in widgets on other unauthed pages (with links to authed tools) that will want the same verify-based functionality. E.g., the unauthed pages for SM, Meds, etc.

Possible tasks:
- 

## Acceptance criteria

- 
- QA review
- A11y review

**Comment by sterkenburgsara** [comment]
Update: Response in this [Slack thread](https://dsva.slack.com/archives/C01K37HRUAH/p1721238858265249) asserts a definitive plan for handling ID-verification globally across VA.gov **before** allowing users to get to apps that require it. 

<img width="762" alt="Screenshot 2024-07-17 at 1 56 44 PM" src="https://github.com/user-attachments/assets/f2a2b720-d8e8-4136-9c04-f3b2816a3d24">

**Comment by sterkenburgsara** [comment]
For future-state evolution of static pages for health tools such as Secure Messages, Medications, and Medical Records, the second comment that Mikki made above will apply. Additional MHV-specific data checks will need to be met. 

We'll need to reveal links to My HealtheVet on these pages after the sign-in and ID-verification checks have been met & let users access `/my-health`, but they'll need to see an additional "Register with My HealtheVet" alert if they don't pass the that check. 

Options for this additional MHV-API check become: 
* Route-guarding users up to the MHV landing page to see the "Register with My HealtheVet" alert
* Letting the tool team apps serve this alert up on their own pages.

**Comment by wesrowe** [comment]
@sterkenburgsara, to clarify – the sign-in widget will NOT check facility registrations, right? If a user is identity-verified, the widget will link them to the appropriate destination, and it will be up to the destination app to tell them, e.g., they don't have access to MHV.

**Comment by sterkenburgsara** [comment]
@wesrowe correct - the sign-in widget will only do two checks: 
* Initial sign-in 
* ID-verification check (must have an LOA3)

Any other checks should happen within the portal, after letting a user navigate to `/my-health `

**Comment by sterkenburgsara** [comment]
The pattern of 1.) initial sign-in and 2.) id-verification check will be needed for **all unauth entry points to anything in the health portal.** 

Very reusable.

**Comment by carlosfelix2** [comment]
Here are are findings and suggestions for adding this content/widget:
1. The CTA widget is a large collection of alerts used for static pages. It has a lot of conditional logic that makes the code hard to follow and maintain. We recommend creating a new static page component to solve the specific needs of health app sign in alerts.
2. This new static page component will follow the same pattern as the other static page components. It will use a new ID on the static pages. Beyond that, the inclusion of this new component is identical to that of the CTA widget.
3. Care must be taken so that this new component does not end up as complicated as the CTA widget. We expect that we will have to use the appID to display different alerts, but we must resist the urge to add conditional logic beyond the intent of alerts.

**Comment by wesrowe** [comment]
@carlosfelix2, I'm curious how the SM team implemented their widget with an alert. Is it inside that jumble of code you referenced?

**Comment by carlosfelix2** [comment]
@wesrowe Yes, it is [here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/static-pages/cta-widget/components/messages/NoMHVAccount.jsx).

**Comment by carlosfelix2** [comment]
Further suggestion on the component's content:
The content of the alert is dependent on the status of the user's credentials. At this time, there are only three versions of the alert [per this article](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/content-patterns-and-standards/sign-in-alerts.md#verify-alert-for-loa1ial1-logingov-and-idme-accounts):
1. Not signed in
2. User is LOA1 (not verified) for Login.gov and ID.me accounts
3. Sign in with a different account for users with non Login.gov or ID.me accounts

This means we can create a generic component that does not require app specific content and hence will be simpler than the CTA widget.

**Comment by radavis** [comment]
Closing since Carlos created the implementation ticket
